portfolio/src/main/java/com/google/sps/data/Comment.java

```diff
@@ -14,17 +14,68 @@
 
 package com.google.sps.data;
 
+import com.google.appengine.api.datastore.Entity;
+import com.google.sps.helper.ValidationResult;
+import com.google.sps.servlets.DataServlet;
+import javax.servlet.http.HttpServletRequest;
+
 /** Class containing comment data. */
 public final class Comment {
 
+  // Whitelist to avoid the possibility of XSS
+  private static final String UNSAFE_CHARACTERS_REGEX = "[^A-Za-z0-9._~()'!*:@,;+?\\s-]";
+
   public static final String AUTHOR_KEY = "author";
   public static final String BODY_KEY = "comment-body";
   public static final String TIME_POSTED_KEY = "timePosted";
+  
+  /**
+   * Get and validate a comment object from an incoming request.
+   * @param request
+   * @return Returns a new comment object or a validation error message.
+   */
+  public static ValidationResult<Comment> getIncomingComment(HttpServletRequest request) {
+    String commentAuthor = DataServlet.getParameter(request, AUTHOR_KEY, "");
+    String commentBody = DataServlet.getParameter(request, BODY_KEY, "");
+
+    commentAuthor = commentAuthor.replaceAll(UNSAFE_CHARACTERS_REGEX, "");
+    commentBody = commentBody.replaceAll(UNSAFE_CHARACTERS_REGEX, "");
+
+    String validationError = validateIncomingComment(commentAuthor, commentBody);
+    if (validationError != null) {
+      return new ValidationResult<Comment>(validationError);
+    } else {
+      Comment newComment = new Comment(commentAuthor, commentBody);
+      return new ValidationResult<Comment>(newComment);
+    }
+  }
+
+  /**
+   * Validates the comment parameters of a new comment.
+   * @return Return the error string or null if there are no errors.
+   */
+  private static String validateIncomingComment(String commentAuthor, String commentBody) {
+    String validationErrors = "";
+
+    if (commentAuthor.isBlank()) {
+      validationErrors += "Please include a comment author (cannot be whitespace).";
+    }
+    if (commentBody.isBlank()) {
+      if (validationErrors.length() > 0) validationErrors += " ";
```

Comment:
```suggestion
    if (commentAuthor.isBlank()) {
      validationErrors += "Please include a comment author (cannot be whitespace).";
    }
    if (commentBody.isBlank()) {
      if (validationErrors.length() > 0) validationErrors += " ";
```
    if (commentAuthor.isBlank()) {
      validationErrors += "Please include a comment author (cannot be whitespace). ";
    }
    if (commentBody.isBlank()) {

Reply:
Okay. I'm not sure if I know where you moved the space to? I might just be having trouble with the suggestion thing.